portfolio/src/main/webapp/script.js

```diff
@@ -12,6 +12,47 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+/**
+ * Randomly selects a background color when page refreshes.
+ */
+function setBackgroundColor() {
+  const bgColors = ['#1A535C', '#4ECDC4', '#F7FFF7', '#A82431', '#FFE66D'];
+
+  // Pick a random hex code from bgColors
+  const bgColor = bgColors[Math.floor(Math.random() * bgColors.length)];
+  
+  // Change the background color of page to bgColor.
+  document.body.style.backgroundColor = bgColor;
+
+  // Adjust text color depending on background color selected.
+  setTextColor(bgColor);
+}
+
+/**
+ * Modifies the color of text depending on the background color, for visibility. 
+ * Uses W3 color contrast standard for calculations https://www.w3.org/TR/AERT/#color-contrast
+ */
+function setTextColor(bgColor) {
```

Comment:
There is a tradeoff you can make here. Curious to hear your thoughts about which you prefer.

Right now you call setTextColor, which parses bgColor and runs the colorDifference formula every time the page loads. 

But the text color depends only on the bg color. Since there are only 5 possible bg colors, there are also only 5 possible fg colors. This means it is possible to "precompute" the bgcolor and fgcolor pairs and avoid this computation at runtime. That implementation would look something like this:

```
const bgColors = [bg1, bg2, bg3, ...];
const fgColors = [fg1, fg2, fg3, ...];
const colorPair = Math.floor(Math.random() * bgColors.length);
setBgColor(bgColors[colorPair]);
setFgColor(fgColors[colorPair]);
```

what do you think?